backend/src/bundles/users/user.repository.ts

```diff
@@ -158,6 +174,122 @@ class UserRepository implements Repository {
     public delete(): ReturnType<Repository['delete']> {
         return Promise.resolve(true);
     }
+
+    public async addFollowing(
+        id: number,
+        followingId: number,
+    ): Promise<UserFollowingsResponseDto | null> {
+        const trx = await this.userModel.startTransaction();
+        try {
+            const user = await this.userModel.query(trx).findById(id);
+            if (!user) {
+                return null;
+            }
+            await user
+                .$relatedQuery('userFriends', trx)
+                .insert({ followingId })
+                .returning('*')
+                .first();
+
+            const userDetails = await this.userModel
+                .query(trx)
+                .findById(followingId)
+                .select('email', ...USER_DETAILS_SELECT_FIELDS)
+                .leftJoin('user_details as ud', `ud.${USER_ID}`, 'users.id');
+
+            await trx.commit();
+
+            return userDetails as unknown as UserFollowingsResponseDto;
+        } catch (error) {
+            await trx.rollback();
+            throw new Error(`Error adding user friend: ${error}`);
+        }
+    }
+
+    public async removeFollowing(
+        id: number,
+        followingId: number,
+    ): Promise<number> {
+        const trx = await this.userModel.startTransaction();
+        try {
+            const user = await this.userModel.query(trx).findById(id);
+            if (!user) {
+                return 0;
```

Comment:
why return 0?

Reply:
Thanks, now in this case I throw an error